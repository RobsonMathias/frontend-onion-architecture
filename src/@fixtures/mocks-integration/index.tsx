export * from './store.mock';
export * from './adapters.mock';
export * from './../mocks/http.mock';
export * from './../mocks/router.mock';
