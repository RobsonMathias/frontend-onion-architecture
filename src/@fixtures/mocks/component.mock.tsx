import React from 'react';

export function Mock() {
  return <div />;
}

export const mockComponent = (componentsName: string[] = []) => {
  let composed: { [key: string]: React.ReactNode } = {};
  componentsName.forEach((name: string) => {
    composed[name] = Mock;
  });
  return composed;
};
