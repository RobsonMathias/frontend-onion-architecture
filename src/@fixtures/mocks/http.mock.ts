export const mockHttp = {
  patch: jest.fn(),
  get: jest.fn(),
  post: jest.fn(),
  put: jest.fn(),
  delete: jest.fn(),
  interceptors: {
    request: {
      use: jest.fn(),
      eject: jest.fn()
    },
    response: {
      use: jest.fn()
    }
  }
} as any;
