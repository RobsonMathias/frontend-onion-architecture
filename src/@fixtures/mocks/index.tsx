export * from './store.mock';
export * from './router.mock';
export * from './component.mock';
export * from './wrappers';
export * from './http.mock';
export * from './services.mock';
export * from './navigation.mock';
export * from './storage.mock';
export * from './use-auth.mock';
