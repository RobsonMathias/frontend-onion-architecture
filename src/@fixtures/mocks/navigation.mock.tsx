import React from 'react';

const setData = jest.fn();
const mockUseUI = jest.fn(() => ({ setData }));

const query = { get: jest.fn(), set: jest.fn() };
const set = jest.fn();
const setAll = jest.fn();
const mockUseQuery = jest.fn(() => ({ query, setAll, set }));

const replace = jest.fn();
const mockUseURL = jest.fn(() => ({ replace }));

const mockNavBar = React.forwardRef((props, ref) => <div {...props} />);

export const mockNavigation = {
  NavBar: mockNavBar,
  useQuery: mockUseQuery,
  query,
  setAll,
  set,
  useUI: mockUseUI,
  setData,
  useURL: mockUseURL,
  replace
};
