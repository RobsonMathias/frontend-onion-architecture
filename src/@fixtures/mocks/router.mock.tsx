import React from 'react';

function Mock({ children }: any) {
  return <>{children}</>;
}

const mockUseLocation = jest.fn(() => ({}));
const goBack = jest.fn();
const push = jest.fn();
const listen = jest.fn();
const mockUseHistory = jest.fn(() => ({ push, goBack, location: {}, listen, action: 'ACTION' }));
const useRouteMatch = () => ({ push: jest.fn(), url: '/mocked' });
const mockUseParams = jest.fn();
const mockNavLink = React.forwardRef(
  ({ to, children, activeClassName, className, 'data-testid': testId }: any, ref) => (
    <a
      className={`${className}-${activeClassName}`}
      href={to}
      children={children}
      data-testid={testId}
    />
  )
);
const Link = React.forwardRef((props, ref) => <p {...props} />);
const mockRoute = React.forwardRef((props, ref) => <div />);
const mockSwitch = React.forwardRef((props, ref) => <div />);
jest.mock('react-router-dom', () => ({
  Redirect: Mock,
  BrowserRouter: Mock,
  useHistory: mockUseHistory,
  useParams: mockUseParams,
  useRouteMatch,
  NavLink: mockNavLink,
  Route: mockRoute,
  Switch: mockSwitch,
  useLocation: mockUseLocation
}));

export const mockRouter = {
  useHistory: mockUseHistory,
  Link,
  useParams: mockUseParams,
  NavLink: mockNavLink,
  useLocation: mockUseLocation,
  goBack,
  push,
  listen
};
