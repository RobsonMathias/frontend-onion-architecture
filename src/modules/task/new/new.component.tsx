import React, { useEffect } from 'react';

import { TaskPayload } from '@types';
import { FieldErrors, useForm } from 'react-hook-form';
import { Button, Grid, TextField } from '@mui/material';
import { BiPlus } from 'react-icons/bi';

export type NewComponentProps = {
  onSubmit: (payload: TaskPayload) => void;
  loading?: boolean;
};

export function NewComponent(props: NewComponentProps) {
  const { onSubmit, loading } = props;

  const {
    register,
    handleSubmit,
    reset,
    formState: { errors, isSubmitted }
  } = useForm();

  function hasError(error?: FieldErrors): string {
    const { message } = error || { message: '' };
    return isSubmitted ? message : '';
  }

  function submit(data: TaskPayload) {
    onSubmit(data);
    reset();
  }

  useEffect(() => {
    return () => {
      reset();
    };
  }, []);

  return (
    <form onSubmit={handleSubmit(submit)}>
      <Grid container>
        <TextField
          variant="filled"
          type="text"
          inputProps={{ 'data-testid': 'description' }}
          label="Insira uma atividade"
          error={!!hasError(errors.description)}
          helperText={hasError(errors.description)}
          {...register('description', {
            required: 'DescriÃ§Ã£o Ã© obrigatÃ³ria.'
          })}
          sx={{ flexGrow: 1 }}
        />
        <Button
          aria-label="Adicionar"
          disabled={loading}
          data-testid="submit"
          type="submit"
          variant="contained"
          color="secondary"
        >
          <BiPlus size={24} />
        </Button>
      </Grid>
    </form>
  );
}
