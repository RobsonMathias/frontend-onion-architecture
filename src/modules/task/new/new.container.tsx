import React from 'react';

import { useTask } from '../hooks';
import { NewComponent } from './new.component';
import { TaskPayload } from '@types';
import { STORE_STATUS_LOADING } from 'constants/store';

export function NewContainer() {
  const { task, create } = useTask();

  function handleSubmit(payload: TaskPayload) {
    create(payload);
  }

  return <NewComponent onSubmit={handleSubmit} loading={task.status === STORE_STATUS_LOADING} />;
}
