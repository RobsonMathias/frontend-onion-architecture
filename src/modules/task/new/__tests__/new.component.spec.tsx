/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';

import { act, fireEvent, render } from '@testing-library/react';
import '@fixtures/mocks';
import React from 'react';

import { NewComponent } from '../new.component';
import { setValueForm } from 'modules/task/new/__tests__/fixture';

const onSubmit = jest.fn();

describe('NewComponent', () => {
  it('should render successfully', () => {
    const { container } = render(<NewComponent onSubmit={onSubmit} />);
    expect(container).toMatchSnapshot();
  });

  it('should submit form successfully', async () => {
    const { getByTestId } = render(<NewComponent onSubmit={onSubmit} />);

    await act(async () => {
      setValueForm(fireEvent, getByTestId, { description: 'new description' });
    });

    expect(onSubmit).toHaveBeenCalledWith({
      description: 'new description'
    });
  });

  it('should render required errors', async () => {
    const { getByTestId, container } = render(<NewComponent onSubmit={onSubmit} />);

    await act(async () => {
      setValueForm(fireEvent, getByTestId, { description: '' });
    });

    expect(container).toHaveTextContent('Descrição é obrigatória.');
  });
});
