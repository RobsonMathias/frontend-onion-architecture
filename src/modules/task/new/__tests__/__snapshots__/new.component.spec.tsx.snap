// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewComponent should render successfully 1`] = `
<div>
  <form>
    <div
      class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root css-te7hn0-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary css-e4w4as-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          for="mui-1"
          id="mui-1-label"
        >
          Insira uma atividade
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
            data-testid="description"
            id="mui-1"
            name="description"
            type="text"
            value=""
          />
        </div>
        
      </div>
      <button
        aria-label="Adicionar"
        class="MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-zcbmsk-MuiButtonBase-root-MuiButton-root"
        data-testid="submit"
        tabindex="0"
        type="submit"
      >
        <svg
          fill="currentColor"
          height="24"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 11h-6V5h-2v6H5v2h6v6h2v-6h6z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </form>
</div>
`;
